python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py

```diff
@@ -552,26 +551,16 @@ def save(self,
                `tf.saved_model.SaveOptions` object that specifies options for
                saving to SavedModel.
         """
-        # get current trained model
-        model = self.get_model()
-
-        if is_local_path(filepath):
-            model.save(filepath, overwrite, include_optimizer, save_format, signatures, options)
-        else:
-            file_name = os.path.basename(filepath)
-            temp_dir = tempfile.mkdtemp()
-            temp_path = os.path.join(temp_dir, file_name)
-            try:
-                model.save(temp_path, overwrite, include_optimizer, save_format, signatures,
-                           options)
-                if save_format == 'h5' or filepath.endswith('.h5') or filepath.endswith('.keras'):
-                    # hdf5 format
-                    put_local_file_to_remote(temp_path, filepath)
-                else:
-                    # tf format
-                    put_local_dir_tree_to_remote(temp_path, filepath)
-            finally:
-                shutil.rmtree(temp_dir)
+        params = dict(
+            filepath=filepath,
+            overwrite=overwrite,
+            include_optimizer=include_optimizer,
+            save_format=save_format,
+            signatures=signatures,
+            options=options
+        )
+        model_refs = [w.save_model.remote(**params) for w in self.remote_workers]
+        ray.get(model_refs[0])
```

Comment:
Just use `ray.get(self.remote_workers[0].save_model.remote(**params))`?

Reply:
We need to save models across all workers, as keras doc mentioned: `You need to save and checkpoint on all workers instead of just one. This is because variables whose synchronization=ON_READ triggers aggregation during saving. It's recommended to save to a different path on each worker to avoid race conditions. Each worker saves the same thing.`